lib/temporal/client/grpc_client.rb

```diff
@@ -97,6 +99,10 @@ def start_workflow_execution(
         end
 
         client.start_workflow_execution(request)
+      rescue GRPC::AlreadyExists => e
+        # Feel like there should be cleaner way to do this...
+        run_id = e.details[/RunId: (.*)\.$/, 1]
+        raise Temporal::WorkflowExecutionAlreadyStartedFailure.new(e.details, run_id)
```

Comment:
I actually wonder if it's helpful to have `run_id` provided here

Reply:
It's useful for singleton workflows where you wish to ensure the workflow is started but it may already be running. Without having `run_id` returned as part of the exception you would have to then send another request to find the `run_id` if you wish to send a signal or so forth.

```
def self.start
  Temporal.start_workflow(
    ...
    options: { workflow_id: 'my-workflow' }
   )
rescue Temporal::WorkflowExecutionAlreadyStartedFailure => e
  e.run_id
end
```

Reply:
Ah, good point 👍